### ISSUE TITLE
多线程训练日志写入异常

### ISSUE THREAD

**Issue by unihornWwan** [PaddlePaddle/PaddleSeg#3007]
### 问题确认 Search before asking

- [X] 我已经查询[历史issue](https://github.com/PaddlePaddle/PaddleSeg/issues)(包括open与closed)，没有发现相似的bug。I have searched the [open and closed issues](https://github.com/PaddlePaddle/PaddleSeg/issues) and found no similar bug report.

### Bug描述 Describe the Bug

您好，当我进行单机多卡训练时，多次遇到训练无故停止的问题，表现为日志在某一时刻不再更新，训练不再继续，但未出现任何 error 提示，GPU 仍处于占用状态，这种情况通常在结束某一轮验证后发生。数据集是自己生成的，脏数据的可能性已被排除，猜测可能为多线程日志写入时的异常，请问是否有相关的解决方法呢，谢谢。

运行代码：

‘’‘
python -m paddle.distributed.launch tools/train.py \
       --config configs/ppmatting/configs/ppmatting/ppmatting-hrnet_w18-human_512.yml \
       --batch_size 12 \
       --iters 131000 \
       --do_eval \
       --save_interval 2620 \
       --num_workers 2 \
       --save_dir output/1432-hr-512-4-gpu-w-rssn-resume-1425 \
       --log_iters 1 \
       --resume_model output/1425-debug-on-loss-frm-1558/iter_73360 \
       --opts train_dataset.dataset_root=dataset/maliang/matting/removebg_portrait_off \
                val_dataset.dataset_root=dataset/maliang/matting/removebg_portrait_off \
                train_dataset.train_file=train_rssn.txt \
                val_dataset.val_file=val_rssn.txt \
                model.backbone.pretrained=pretrained/hrnet_w18_ssld.pdparams \
                lr_scheduler.learning_rate=0.001 \
’‘’

### 复现环境 Environment

OS：Ubuntu
cuda ：11.2
paddlepaddle-gpu ：2.4.1.post112
paddleseg ： 0.0.0.dev0

### Bug描述确认 Bug description confirmation

- [X] 我确认已经提供了Bug复现步骤、代码改动说明、以及环境信息，确认问题是可以复现的。I confirm that the bug replication steps, code change instructions, and environment information have been provided, and the problem can be reproduced.

### 是否愿意提交PR？ Are you willing to submit a PR?

- [X] 我愿意提交PR！I'd like to help by submitting a PR!

**Comment by Stinky-Tofu** [comment]
> ### 问题确认 Search before asking
> * [x]  我已经查询[历史issue](https://github.com/PaddlePaddle/PaddleSeg/issues)(包括open与closed)，没有发现相似的bug。I have searched the [open and closed issues](https://github.com/PaddlePaddle/PaddleSeg/issues) and found no similar bug report.
> 
> ### Bug描述 Describe the Bug
> 您好，当我进行单机多卡训练时，多次遇到训练无故停止的问题，表现为日志在某一时刻不再更新，训练不再继续，但未出现任何 error 提示，GPU 仍处于占用状态，这种情况通常在结束某一轮验证后发生。数据集是自己生成的，脏数据的可能性已被排除，猜测可能为多线程日志写入时的异常，请问是否有相关的解决方法呢，谢谢。
> 
> 运行代码：
> 
> ‘’‘ python -m paddle.distributed.launch tools/train.py --config configs/ppmatting/configs/ppmatting/ppmatting-hrnet_w18-human_512.yml --batch_size 12 --iters 131000 --do_eval --save_interval 2620 --num_workers 2 --save_dir output/1432-hr-512-4-gpu-w-rssn-resume-1425 --log_iters 1 --resume_model output/1425-debug-on-loss-frm-1558/iter_73360 --opts train_dataset.dataset_root=dataset/maliang/matting/removebg_portrait_off val_dataset.dataset_root=dataset/maliang/matting/removebg_portrait_off train_dataset.train_file=train_rssn.txt val_dataset.val_file=val_rssn.txt model.backbone.pretrained=pretrained/hrnet_w18_ssld.pdparams lr_scheduler.learning_rate=0.001 ’‘’
> 
> ### 复现环境 Environment
> OS：Ubuntu cuda ：11.2 paddlepaddle-gpu ：2.4.1.post112 paddleseg ： 0.0.0.dev0
> 
> ### Bug描述确认 Bug description confirmation
> * [x]  我确认已经提供了Bug复现步骤、代码改动说明、以及环境信息，确认问题是可以复现的。I confirm that the bug replication steps, code change instructions, and environment information have been provided, and the problem can be reproduced.
> 
> ### 是否愿意提交PR？ Are you willing to submit a PR?
> * [x]  我愿意提交PR！I'd like to help by submitting a PR!

你好，辛苦提供一下train log

**Comment by unihornWwan** [comment]
您好， train_log 如下

链接：https://pan.baidu.com/s/1QyNPWvd90twA_VImPkWNAg?pwd=4q6c 
提取码：4q6c

**Comment by TingquanGao** [comment]
Thanks for this issue. As it has been inactive for a long time, we would close it. If you has any questions, please feel free to reopen or new issue, and we will follow up and resolve it.
